### DISCUSSION TITLE
Is there a function to read elevation (z) at a certain point (x,y) ?

### DISCUSSION THREAD

**Discussion by valentinMachado** [iTowns/itowns#1872] [Q&A] [answered]
Hi,

I am trying to make a character walk over the ground (planar THREE.Object ) which has an elevation layer in a PlanarView. I have tried to raycast the object from above looking down the z axis but this return me always 0 value. I checked the geometry of the planar THREE.Object and found for every point a 0 in the z coordinate. I assume that the elevation layer map is passed at vertex shader as a displacement map and move the vertex only on the GPU side, am I wrong ?

My question is how do I get the value on the CPU side to determine the elevation of the object moving on the ground ?

First time here tell me if this is the good place to post this.
Thanks !

**Comment by jailln** [comment, marked as answer]
Hello @valentinMachado,

You can use [DEMUtils](http://www.itowns-project.org/itowns/docs/#api/Geographic/DEMUtils) `getElevationValueAt` method.

**Reply by valentinMachado** [reply]
Hello @jailln 
This looks like exactly what I need I'll give it a try !
Thank you :)

**Reply by valentinMachado** [reply]
It worked just fine !
